docs/DeveloperGuide.md

```diff
@@ -6,8 +6,248 @@
+customer name `Bob` and satisfaction rating `5`.
+
+![Step 1 Object Diagram](team/alinazheng_addsatisfactioncommand_uml/AlinaZheng_AddSatisfaction_Step4ObjectDiagram.png)
+
+Step 5. When the `AddSatisfactionCommand` object is executed in the `Duke` class, the `Satisfaction` object
+created in the `AddSatisfactionCommand` class is added to `satisfactionList`.
+
+![Step 1 Object Diagram](team/alinazheng_addsatisfactioncommand_uml/AlinaZheng_AddSatisfaction_Step5ObjectDiagram.png)
+
+The following sequence diagram shows what would happen if the user typed `add satisfaction Bob 5`.
+
+PLEASE NOTE: Due to the limitation of PlantUML, the lengths of the activation bars may not be correct, and
+certain return values from method calls that could not be placed correctly were omitted. For example, the
+instantiation of objects of type `SatisfactionList`, `CommandParser`, `Command`, `AddSatisfactionCommand`,
+and `Satisfaction` should have return value arrows (representing the object that was instantiated) to the
+entity that created those instances, but those are not indicated in the diagram. Furthermore, for neatness, the
+`User` entity should be the left-most entity, but that could not be formatted properly in the diagram.
+
+![Sequence Diagram](team/alinazheng_addsatisfactioncommand_uml/AlinaZheng_AddSatisfaction_SequenceDiagram.png)
+
+### Add Housekeeper feature
+
+
+The add housekeeper mechanism is facilitated by `AddHousekeeperCommand`. It extends command. Additionally, 
+it implements the following operations:
+* `AddHousekeeperCommand#extractDetails()`— Stores the name and age of the new Housekeeper
+* `AddHousekeeperCommand#extractName()`— Derive the name of the Housekeeper
+* `AddHousekeeperCommand#extractAge()`— Derive the age of Housekeeper and cast it into an integer
+* `AddHousekeeperCommand#execute()` — Executes the addition of new housekeeper into list
+
+Given below is an example usage scenario of how AddHousekeeperCommand behaves at each step.
+
+Step 1: User launches the application for the first time. The list of housekeeper in records is empty.
+
+Step 2: User give an add housekeeper command `add housekeeper susan / 33`. The input will be split by the delimiter
+`/` to derive `susan` and `33`. The `CommandParser` runs parse which will return a new `AddHousekeeperCommand`.
+
+Step 3: `AddHousekeeperCommand#extractDetails()` will call `AddHousekeeperCommand#extractName()` 
+and `AddHousekeeperCommand#extractAge()` to derive `susan` and `33`.
+
+Step 4: The `AddHousekeeperCommand` will now contain `susan` and `33`.
+
+Step 5: The execute method will derive the housekeeper name, `susan` which would call `HousekeeperList#hasNameAdded()` 
+to verify if name has been recorded. If name is not in records, `HousekeeperList#addHousekeeper()` would be called to 
+add the housekeeper, `susan` and `33` into the housekeeperList.
+
+![Sequence](team/falicia_addHousekeeperCommand/sequenceAddHousekeeperCommandv2.jpg)
```

Comment:
![image](https://user-images.githubusercontent.com/69459230/160963952-8d5a898d-d26e-4cfb-9276-58848e119dc0.png)

Consider aligning all arrows (e.g. arrowhead for methods should point to the tip of the activation bar) and offsetting each separate arrow to indicate order of occurrence (e.g. parse() occurs before AddHousekeeperCommand() and thus the arrow for AddHousekeeperCommand() should be offset slightly below the arrow of parse().